[docs] for data storage in neo

im-xra-dev opened the pull request:
I have written up the initial documentation for the data layout in the neo store. It represents the connections and nodes required to provide relevant content to users

im-xra-dev commented on docs/neo.md:
```diff
@@ -0,0 +1,123 @@
+
+### Recipe
+
+Each recipe is a subtype of a post. It contains the attributes and relationships of the Post nodes, as well as
```
```suggestion
Each recipe is a subtype of a post. It contains the attributes and relationships of the Post nodes.
```

im-xra-dev commented:
lgtm

DanielShandley commented on docs/neo.md:
```diff
@@ -0,0 +1,123 @@
+to try at some point
+
+_Attributes_
+
```
Something that I think should be a core feature is the ability to rate recipes, my parents check their cookbooks for recipes that they have personally tried and rated highly, I think having some kind of star system would be beneficial.

DanielShandley reviewed (changes requested):
Le data structure image plez :)

im-xra-dev commented:
I have updated the docs to contain a number ranking as opposed to boolean favourite for recipes. I have also added a relationship for users and recipe-tags with a score attribute to further allow us to gain insight to preferences of the user.

I realised there was no Authored relation between users and content so I added that, along side an example data image (this will only appear embedded into the actual docs file once merged into main but is available [here](https://github.com/im-xra-dev/chomp-timeline/blob/neo-docs/docs/neo.png) for preview)

DanielShandley reviewed (approved):
All looks good to me.
